File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        unordered_map<int,int> mp; 
        for(int i=0;i<nums.size();i++)
        {
            mp[nums[i]]++;
        } int ans=0;
        for(auto i:mp)
        {
            if(i.second>1) ans=i.first;
        }
        return ans;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        HashSet<Integer> st=new HashSet<>(); 
        for(int i=0;i<nums.length;i++)
        {
            if(st.contains(nums[i]))
            {
                return nums[i];
            }
            st.add(nums[i]); 
        }
        return -1;
    }
}